pkg/controller/instance/delta/delta_test.go

```diff
@@ -458,3 +458,121 @@ func TestCompare_EmptyMaps(t *testing.T) {
 		})
 	}
 }
+
+func TestCompare_EmptyArraysAndNulls(t *testing.T) {
+	tests := []struct {
+		name     string
+		desired  *unstructured.Unstructured
+		observed *unstructured.Unstructured
+		wantDiff bool
+	}{
+		{
+			name: "empty array equals null in nested structure",
```

Comment:
minor:
```suggestion
			name: "empty array equals nil in nested structure",
```

and below